File: Biopython/parse_mmcif.py
# Benchmark the parsing of a mmCIF file given as an argument

import sys
import time
from Bio.PDB import MMCIFParser

mmcif_filepath = sys.argv[1]
parser = MMCIFParser()

start = time.time()
parser.get_structure("", mmcif_filepath)
end = time.time()

print(end - start)

File: BioStructures/parse_mmcif.jl
# Benchmark the parsing of a mmCIF file given as an argument

using BioStructures

mmcif_filepath = ARGS[1]

# Run to JIT compile
read(mmcif_filepath, MMCIF)

elapsed = @elapsed struc = read(mmcif_filepath, MMCIF)

println(elapsed)
